src/lib/polkadotApi.ts

```diff
@@ -1,27 +1,47 @@
 import { ApiPromise, Keyring, WsProvider } from '@polkadot/api';
 import { getBlockNumber, getEpoch } from './connections';
-import { dotApi, storeBlockNumber, storeConnected, storeToken } from './stores';
+import { dotApi, storeChainInfo } from './stores';
 import { options } from '@frequency-chain/api-augment';
 
-import type { DotApi, MsaInfo } from '$lib/storeTypes';
+import type { ChainInfo, DotApi, MsaInfo } from '../lib/storeTypes';
 import type { KeyringPair } from '@polkadot/keyring/types';
 import type { ChainProperties } from '@polkadot/types/interfaces';
 import type { Option, u64 } from '@polkadot/types';
 
 export type AccountMap = Record<string, KeyringPair>;
 
-export async function getApi(selectedProviderURI: string, thisDotApi: DotApi, wsProvider: WsProvider) {
+export async function updateConnectionStatus(apiPromise: ApiPromise) {
+  const chain = await apiPromise.rpc.system.properties();
+  const token = getToken(chain);
+  const blockNumber = (await getBlockNumber(apiPromise)) as bigint;
+  const epochNumber = await getEpoch(apiPromise);
+  storeChainInfo.update((info: ChainInfo) => (info = { connected: true, blockNumber, epochNumber, token }));
+}
+
```

Comment:
This should be an API class with methods of connect(), disconnect(), etc...

Reply:
We only have one connection and a secondary temporary connection. Can you please elaborate more on why you think we need a class?

Reply:
It's optional but it's a best practice for reusability and maintenance.

Reply:
Thoughts? @JoeCap08055

Reply:
Hmmm, seems like philosophical differences. Being an OO guy, I lean toward encapsulation. The current use case is that there is a singleton in the app, but what if that changes? Classes make it easy to associate state and data with a particular instance.

From a logical code organization standpoint, if I have a bunch of methods that are closely associated in function/purpose, I at least like to put them into a namespace for grouping, rather than relying on file structure to communicate association.

But, that might just be my old-school OO way of thinking...

The current implementation is very Javascript-y; but I thought one of the purposes of Typescript was to introduce classes and other types... so maybe classes are more Typescript-y?

I think this is more a question of what's more consistent with the design methodology in the rest of the project, rather than a right/wrong way to do it.